### PR TITLE
Update pipelines.adoc

### PR DESCRIPTION
**Pull request by aadeshpa**
updated the document with the configmap and secret for setting the custom collection registry url

### CODE REVIEW

File: pipelines.adoc [hunk 1]
```diff
@@ -159,23 +159,57 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+== Execute pipelines manually from the command line for your custom built Collections
 
-You can check the status of the pipeline run from the OKD console,
-command line, or Tekton dashboard.
+If you have your own custom build collection image and want to run any pipelines against it instead of out of box kabanero collections, then you need to follow below steps as per the required scenarios.
 
+=== Setup of container registry url for the custom Collection image
 
-===  Ashish Execute pipelines manually from the command line for your custom built Collections
+By default the pipelines assume you are pulling the collection from `kabanero` so it looks for the collection image in dockerhub repository. If you want to use your container registry URL where you have the collection image you would need to follow below steps
```
Comment (kvijai82) [inline pipelines.adoc]: I would reword it to maybe something like this.  It's good from most part, but the statement it pulls from kabanero is not fully true.  It's more that it looks in docker hub.

By default, the pipelines pull the collection images for docker hub.  If you are publishing your collection images to any other repository, follow the instructions below to configure the custom repository that the pipelines should pull the collection images from.

File: pipelines.adoc [hunk 2]
```diff
@@ -159,23 +159,57 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+ cd kabanero-pipelines/pipelines/common/
+ vi collection-image-registry-map.yaml
+
+. Apply the above configmap file which will set your container registry.
+
+ oc apply -f collection-image-registry-map.yaml
+
+==== For Collection image stored in container registry with internal route url on the OCP cluster
+
+For internal openshift registry you need to only setup above `collection-image-registry-map.yaml` with the internal registry url.
```
Comment (kvijai82) [inline pipelines.adoc:182]: Also maybe add that the service account should let the pipelines pull from the registry without needing to configure a secret.

**Review by kvijai82 — APPROVED** [review]

File: pipelines.adoc [hunk 3]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
 oc apply -f <collection-name>-pipeline-run.yaml
 ```
 
-== Checking the status of the pipeline run
+== Execute pipelines manually from the command line for your custom built Collections
```
Comment (dmuelle) [inline pipelines.adoc:162]: **change**
Execute pipelines manually from the command line for your custom built Collections
**to**
Run pipelines from the command line for your custom built collections

File: pipelines.adoc [hunk 4]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
 
-You can check the status of the pipeline run from the OKD console,
-command line, or Tekton dashboard.
+If you have your own custom build collection image and want to run any pipelines against it instead of out of box kabanero collections, then you need to follow below steps as per the required scenarios.
```
Comment (dmuelle) [inline pipelines.adoc:164]: **change**
If you have your own custom build collection image and want to run any pipelines against it instead of out of box kabanero collections, then you need to follow below steps as per the required scenarios.
**to**
The following steps explain how to run pipelines against custom built collection images instead of the provided Kabanero collections.

File: pipelines.adoc [hunk 5]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
 
+=== Setup of container registry url for the custom Collection image
```
Comment (dmuelle) [inline pipelines.adoc:166]: **change**
Setup of container registry url for the custom Collection image
**to**
Set up a container registry URL for the custom collection image

File: pipelines.adoc [hunk 6]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
 
-===  Ashish Execute pipelines manually from the command line for your custom built Collections
+By default, the pipelines pull the collection images for docker hub. If you are publishing your collection images to any other repository, follow the instructions below to configure the custom repository that the pipelines should pull the collection images from.
```
Comment (dmuelle) [inline pipelines.adoc:168]: **change**
By default, the pipelines pull the collection images for docker hub. If you are publishing your collection images to any other repository, follow the instructions below to configure the custom repository that the pipelines should pull the collection images from.

**to**
By default, pipelines pull the collection images for Docker hub. If you are publishing your collection images to any other repository, use the following process to configure the custom repository from which your pipelines pull the collection images.

File: pipelines.adoc [hunk 7]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
 
-If you have your own custom build collection image and want to run any pipelines against it instead of out of box kabanero collections, then you need to follow below steps as per the required scenarios.
+. After you have cloned kabanero-pipelines repository find the configmap template file `collection-image-registry-map.yaml` and add your container registry URL in it for the key `default-collection-image-registry-url`
```
Comment (dmuelle) [inline pipelines.adoc:170]: **change**
After you have cloned kabanero-pipelines repository find the configmap template file `collection-image-registry-map.yaml` and add your container registry URL in it for the key `default-collection-image-registry-url` 

**to**
After you clone the kabanero-pipelines repository, find the `collection-image-registry-map.yaml` configmap template file.  Add your container registry URL to this file in place of the `default-collection-image-registry-url` statement.

File: pipelines.adoc [hunk 8]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
-If you have your own custom build collection image and want to run any pipelines against it instead of out of box kabanero collections, then you need to follow below steps as per the required scenarios.
+. After you have cloned kabanero-pipelines repository find the configmap template file `collection-image-registry-map.yaml` and add your container registry URL in it for the key `default-collection-image-registry-url` 
 
-=== Setup of container registry url for the custom Collection image
 
-. By default the pipelines assume you are pulling the collection from `kabanero` so it looks for the collection image in dockerhub repository. If you want to use
+ cd kabanero-pipelines/pipelines/common/
+ vi collection-image-registry-map.yaml
+
+. Apply the above configmap file which will set your container registry.
```
Comment (dmuelle) [inline pipelines.adoc:176]: **change**
 cd kabanero-pipelines/pipelines/common/
 vi collection-image-registry-map.yaml

. Apply the above configmap file which will set your container registry.

**to**
. Apply the following configmap file, which will set your container registry. 

cd kabanero-pipelines/pipelines/common/
 vi collection-image-registry-map.yaml

File: pipelines.adoc [hunk 9]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+ oc apply -f collection-image-registry-map.yaml
```
Comment (dmuelle) [inline pipelines.adoc:178]: what is the significance of this command? it's not explained

File: pipelines.adoc [hunk 10]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+==== For Collection image stored in container registry with internal route url on the OCP cluster
```
Comment (dmuelle) [inline pipelines.adoc:180]: **change**
For Collection image stored in container registry with internal route url on the OCP cluster
**to**
Set up a container registry URL for a custom collection image that is stored in a container registry with an internal route URL on the OCP cluster

File: pipelines.adoc [hunk 11]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+For internal openshift registry you need to only setup above `collection-image-registry-map.yaml` with the internal registry url.
+
+NOTE : The service account should let the pipelines pull from the internal registry without needing to configure a secret.
```
Comment (dmuelle) [inline pipelines.adoc:184]: **change**
NOTE : The service account should let the pipelines pull from the internal registry without needing to configure a secret.

**to**
In this case, the service account that is associated with the pipelines must be configured to allow the pipelines pull from the internal registry without configuring a secret.

is there somewhere we can point to for information about service accounts? ie "For more information, see x"

File: pipelines.adoc [hunk 12]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+ cd kabanero-pipelines/pipelines/common/
+ vi collection-image-registry-map.yaml
+
+. Apply the above configmap file which will set your container registry.
+
+ oc apply -f collection-image-registry-map.yaml
+
+==== For Collection image stored in container registry with internal route url on the OCP cluster
+
+For internal openshift registry you need to only setup above `collection-image-registry-map.yaml` with the internal registry url.
```
Comment (dmuelle) [inline pipelines.adoc:182]: **change**
For internal openshift registry you need to only setup above `collection-image-registry-map.yaml` with the internal registry url.
**to**
For an internal OpenShift registry, set up the `collection-image-registry-map.yaml` file with the internal registry URL.

File: pipelines.adoc [hunk 13]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+NOTE : The service account should let the pipelines pull from the internal registry without needing to configure a secret.
 
-==== For Collection image stored with internal openshift registry route url on the OCP cluster
+==== For Collection image stored in container registry with external route url
```
Comment (dmuelle) [inline pipelines.adoc:186]: **Change**
For Collection image stored in container registry with external route url
**to**
Set up a container registry URL for a custom collection image that is stored in a container registry with an external route URL

File: pipelines.adoc [hunk 14]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
 
-==== For Collection image stored with external container registry route url
+For collection image with external container registry route url you need to setup a kubernetes secret in addition to above configured configmap as per below steps.
```
Comment (dmuelle) [inline pipelines.adoc:188]: what is the purpose of the kubernetes secret? it's not mentioned again.

File: pipelines.adoc [hunk 15]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+. Find the template file `default-collection-image-registry-secret.yaml` in cloned kabanero-pipelines repo and update it with the username and token password for the container registry url you specified above.
```
Comment (dmuelle) [inline pipelines.adoc:190]: **change**
. Find the template file `default-collection-image-registry-secret.yaml` in cloned kabanero-pipelines repo and update it with the username and token password for the container registry url you specified above.
**to**
. First, update the `collection-image-registry-map.yaml` file with your container registry file, as described in step 1 of `Set up a container registry URL for the custom collection image`.

File: pipelines.adoc [hunk 16]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+ cd kabanero-pipelines/pipelines/common
```
Comment (dmuelle) [inline pipelines.adoc:192]: is this command necessary? should step one here be "Go to the `kabanero-pipelines/pipelines/common` directory?

File: pipelines.adoc [hunk 17]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+ 
+. You need to first create `base64` format of the username and password for the container registry url.
```
Comment (dmuelle) [inline pipelines.adoc:195]: **change**
You need to first create `base64` format of the username and password for the container registry url.
**to**
Create a Base64 format version of the user name and password for the external route container registry url.

File: pipelines.adoc [hunk 18]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+
+ echo -n <your-registry-username> | base64
+ echo -n <your-registry-password> | base64
+ 
+. Update the file `default-collection-image-registry-secret.yaml` with `base64` format from above step output.
```
Comment (dmuelle) [inline pipelines.adoc:200]: **change**
Update the file `default-collection-image-registry-secret.yaml` with `base64` format from above step output.
**to**
Update the `default-collection-image-registry-secret.yaml` file with the Base64 formatted user name and password.

File: pipelines.adoc [hunk 19]
```diff
@@ -159,23 +159,58 @@ A sample `manual-pipeline-run-template.yaml` file is provided in the link:https:
...
+ 
+ vi default-collection-image-registry-secret.yaml
+  
+. Apply the `default-collection-image-registry-secret.yaml` file to the cluster
+
+ oc apply -f default-collection-image-registry-secret.yaml
+
+. Finally run the pipeline as shown in above section.
```
Comment (dmuelle) [inline pipelines.adoc:208]: **change**
Finally run the pipeline as shown in above section.
**to**
You can now run the pipeline by following the steps in the preceding `Run pipelines from the command line for your custom built collections` section.

**Review by dmuelle — CHANGES_REQUESTED** [review]
changes requested below. This will probably require more editing once it goes back to the main repo but I can edit there and send a PR for your approval. Let me know if you have any questions or if I've misunderstood the intention of anything. Thanks

**Review by kvijai82 — APPROVED** [review]

**Review by dmuelle — APPROVED** [review]
approving here to make ID edits in main repo
